components/match2/commons/match_group_input.lua

```diff
@@ -153,27 +165,39 @@ function MatchGroupInput._fetchBracketDatas(templateId, bracketId)
 		return (bracketId or '') .. '_' .. baseMatchId
 	end
 
+	-- Convert 0 based array to 1 based array
+	local function shiftArrayIndex(elems)
+		return Array.extend(elems[0], elems)
+	end
+
 	return Table.map(matches, function(_, match)
 		local bracketData = match.match2bracketdata
-		if String.nilIfEmpty(bracketData.toupper) then
-			bracketData.toupper = replaceBracketId(bracketData.toupper)
-		end
-		if String.nilIfEmpty(bracketData.tolower) then
-			bracketData.tolower = replaceBracketId(bracketData.tolower)
-		end
-		if String.nilIfEmpty(bracketData.thirdplace) then
-			bracketData.thirdplace = replaceBracketId(bracketData.thirdplace)
-		end
-		if String.nilIfEmpty(bracketData.bracketreset) then
-			bracketData.bracketreset = replaceBracketId(bracketData.bracketreset)
-		end
+
+		-- Convert 0 based array to 1 based array
+		bracketData.advanceSpots = bracketData.advanceSpots and shiftArrayIndex(bracketData.advanceSpots)
+		bracketData.lowerEdges = bracketData.lowerEdges and shiftArrayIndex(bracketData.lowerEdges)
```

Comment:
Inconsistent naming when comparing to `bracketData.bracketreset`, etc

Reply:
The convention I'm using is:
- lower case for fields on lpdb objects
- camel case for fields inside json objects (e.g. fields in `extradata`, `stream,` `match2bracketdata.coordinates`, etc)

We've never had a consistent naming scheme for json object fields. For example, both lower case and camel case are used in `match2.extradata`: https://liquipedia.net/starcraft2/Special:LiquipediaDB/Liquipedia:BracketUpdate/Tests/Wide_Load . Because of that inconsistency, I chose to use camel case for newly added fields due to the following reasons:
- It's easier to work with in lua, the primary consumer of lpdb data.
- It allows storing objects directly in lpdb without converting field names. This feature is made use by 
-- `Module:GroupTableLeague/next` which stores the placement results without conversion, and the opponent struct (from Module:Opponent) without conversion.
-- `Module:Bracket/Template` which stores coordinates with minor conversion (just 0 <-> 1 based indexing)

Reply:
Those are reasonable arguments but not for inside a PR. Make them consistent with the existing code, and then if you wish to create a new convention, create a new PR just for that.

Reply:
Okay

Reply:
I changed lowerEdges and advanceSpots but left lowerMatchIds and upperMatchId alone because there's a lot of code using it already.

Reply:
Is that because they were introduced before this PR?

Reply:
Yes. We've been using camel case for extradata/bracketdata fields for several months now